domain/sysvar_cache.go

```diff
@@ -111,7 +112,11 @@ func (svc *SysVarCache) fetchTableValues(ctx sessionctx.Context) (map[string]str
 
 // RebuildSysVarCache rebuilds the sysvar cache both globally and for session vars.
 // It needs to be called when sysvars are added or removed.
+// Only one rebuild can be in progress at a time, this prevents a lost update race
+// where an earlier fetchTableValues() finishes last.
 func (svc *SysVarCache) RebuildSysVarCache(ctx sessionctx.Context) error {
+	svc.rebuildLock.Lock()
+	defer svc.rebuildLock.Unlock()
 	newSessionCache := make(map[string]string)
```

Comment:
These two lines can be moved out of the lock scope.

Reply:
@ngaut did you mean `newSessionCache` and `newGlobalCache`?

I had considered this, but my reason for keeping it protected by the lock is that:
1. The `RebuildSysVarCache` is called asynchronously.
2. The memory footprint will be more stable if concurrent requests don't allocate memory.

It is a small difference however, so I am of course happy to change it.

Reply:
Yes, please. Make sure the lock scope is as small as possible. It is also more clear to avoid reader question why newSessionCache and newGlobalCache are protected by lock.

Reply:
Done